src/core/models/appliedhighlightedfields.js

```diff
@@ -9,48 +8,62 @@ export default class AppliedHighlightedFields {
    * Constructs a highlighted field map which consists of mappings from fields to highlighted
    * value strings.
    *
-   * @param {Object<string, string|object>} data Keyed by fieldName. It may consist of nested fields
+   * @param {Object<string, string|object|array>} data Keyed by fieldName. It may consist of nested fields
    *
    * Example object:
    *
    * {
-   *   name: '<strong>Yext</strong>',
-   *   description: {
-   *     featured: '<strong>Yext</strong> is the offical answers company'
-   *   }
+   *   description: 'likes <strong>apple</strong> pie and green <strong>apple</strong>s',
+   *   c_favoriteFruits: [
+   *     {
+   *       apples: ['Granny Smith','Upton Pyne <strong>Apple</strong>'],
+   *       pears: ['Callery Pear']
+   *     }
+   *   ]
    * }
    */
   constructor (data = {}) {
     Object.assign(this, data);
   }
 
   /**
-   * Constructs an AppliedHighlightedFields object from an answers-core HighlightedValue array
+   * Constructs an AppliedHighlightedFields object from an answers-core HighlightedField
    *
-   * @param {HighlightedValue[]} highlightedValueArray from answers-core
+   * @param {import('@yext/answers-core').HighlightedFields} highlightedFields from answers-core
```

Comment:
nit: if we're using the import syntax then we can probably delete the 'from answers-core' part